superset/connectors/sqla/models.py

```diff
@@ -1425,7 +1425,20 @@ def _get_top_groups(  # pylint: disable=no-self-use
 
         return or_(*groups)
 
+    def get_filter_type(self, filter: dict) -> str:
+        for column in self.columns:
+            if filter['col'] == column.column_name:
+                return column.type
+
     def query(self, query_obj: QueryObjectDict) -> QueryResult:
+        # Converts comparator from string to boolean if field type is boolean
+        for i in range(len(query_obj['filter'])):
+            if self.get_filter_type(query_obj['filter'][i]) == 'BOOLEAN':
+                if query_obj['filter'][i]['val'] == 'true':
+                    query_obj['filter'][i]['val'] = True
+                elif query_obj['filter'][i]['val'] == 'false':
+                    query_obj['filter'][i]['val'] = False
```

Comment:
Does this break boolean operations on other connections like sqlite or PostgreSQL for example?